### PR TITLE
Kernel upgrade script

### PR DESCRIPTION
**Pull request by bhale**
This script behaves similarly to the rootfs upgrade script, but instead upgrades the kernel.

### CODE REVIEW

File: overlay/sbin/k3os-upgrade-kernel [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	echo "This script must be run as root."
+	exit 1
+fi
+
+PROC=$(uname -m)
+
+# Take a parameter of the version number (i.e. v0.4.0) if it is given, otherwise use latest
+if [ -z $K3OS_VERSION ]
+then
+	K3OS_VERSION=$(curl -sL api.github.com/repos/rancher/k3os/releases/latest | jq .tag_name -r)
```
Comment (zimme) [inline overlay/sbin/k3os-upgrade-kernel]: We should be able to use `latest` as the default version, we just need to update the download links. The following link work. https://github.com/rancher/k3os/releases/latest/download/k3os-kernel-version-amd64.
```suggestion
	K3OS_VERSION=latest
```

File: overlay/sbin/k3os-upgrade-kernel [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+if [ $PROC == "x86_64" ]
+then
+	ARCH="amd64"
+elif [ $PROC == "aarch64" ]
+then
+	ARCH="arm64"
+elif [ $PROC == "armv7l" ]
+then
+	ARCH="arm"
```
Comment (zimme) [inline overlay/sbin/k3os-upgrade-kernel]: From what I can tell, there is only a rootfs distribution for `arm` so this script will probably "crash" if run on an arm board. Either we remove this check or use it to inform users about `arm` being a rootfs only distribution.

File: overlay/sbin/k3os-upgrade-kernel [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+else
+	echo "Unsupported CPU architecture."
+	exit 1
+fi
+
+KERNEL_VERSION=$(curl -sL https://github.com/rancher/k3os/releases/download/${K3OS_VERSION}/k3os-kernel-version-${ARCH})
```
Comment (zimme) [inline overlay/sbin/k3os-upgrade-kernel]:
```suggestion
KERNEL_VERSION=$(curl -sL https://github.com/rancher/k3os/releases/${K3OS_VERSION}/download/k3os-kernel-version-${ARCH})
```

File: overlay/sbin/k3os-upgrade-kernel [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+echo "Upgrading k3os kernel to ${KERNEL_VERSION}"
+
+cd /k3os/system/
+mount -o remount rw .
+cd kernel
+mkdir -p ${KERNEL_VERSION}
+cd ${KERNEL_VERSION}
+curl -sL https://github.com/rancher/k3os/releases/download/${K3OS_VERSION}/k3os-initrd-${ARCH} --output initrd
```
Comment (zimme) [inline overlay/sbin/k3os-upgrade-kernel]:
```suggestion
curl -sL https://github.com/rancher/k3os/releases/${K3OS_VERSION}/download/k3os-initrd-${ARCH} --output initrd
```

File: overlay/sbin/k3os-upgrade-kernel [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+curl -sL https://github.com/rancher/k3os/releases/download/${K3OS_VERSION}/k3os-kernel-${ARCH}.squashfs --output kernel.squashfs
```
Comment (zimme) [inline overlay/sbin/k3os-upgrade-kernel]:
```suggestion
curl -sL https://github.com/rancher/k3os/releases/${K3OS_VERSION}/download/k3os-kernel-${ARCH}.squashfs --output kernel.squashfs
```

File: overlay/sbin/k3os-upgrade-kernel [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+cd ..
+
+PREVIOUS_KERNEL=$(ls -l current | awk '{print $11}')
+ln -s ${PREVIOUS_KERNEL} previous
+unlink current
+ln -s ${KERNEL_VERSION} current
+sync
+
+echo "Upgrade complete! Please reboot."
```
Comment (zimme) [inline overlay/sbin/k3os-upgrade-kernel]: As we're not actually rebooting in the script we should probably remount `/k3os/system` as read-only again before we're done.

File: overlay/sbin/k3os-upgrade-kernel (hunk 2 above)
Comment (bhale) [reply overlay/sbin/k3os-upgrade-kernel]: Ah, that makes sense. Will adjust here to exit on arm.

File: overlay/sbin/k3os-upgrade-kernel (hunk 6 above)
Comment (bhale) [reply overlay/sbin/k3os-upgrade-kernel]: I think we should trust the admin ( or future operator ) to reboot after.

File: overlay/sbin/k3os-upgrade-kernel [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+if [ $PROC == "x86_64" ]
+then
+	ARCH="amd64"
+elif [ $PROC == "aarch64" ]
+then
+	ARCH="arm64"
+elif [ $PROC == "armv7l" ]
+then
+	ARCH="arm"
+	echo "K3os for 32-bit ARM architecture only provides rootfs, no kernel updates available."
```
Comment (zimme) [inline overlay/sbin/k3os-upgrade-kernel]:
```suggestion
	echo "k3os for 32-bit ARM architecture only provides rootfs, no kernel updates available."
```
😱

**Pull request merged**